uktrade/data-hub-frontend
# Pipeline Edit

## Description of change
This PR add the ability to edit pipelines. It shares the form component for create and update so there was a small refactor to share the forms.
There was also a refactor to move the pipeline app out of the company subapp into its own app. The reasoning behind this was the company middleware was only used for adding a pipeline. For updating and viewing pipelines there is no need for the company middleware. 

There is also a fix to the form to submit when there is only one step. This fix is on master can not rebase off master because I've branched off feature/pipeline. This will be fixed in a rebase and the commit will be omitted. https://github.com/uktrade/data-hub-frontend/pull/2691/commits/cb56ddb21dd4361408b74cc69517bb306e134861

## Test instructions
on the deployment site go to /my-pipeline/876108b8-32ac-47b3-b641-c4089447a4bc/edit and change the values click on update and you should be redirected to /my-pipelines. Go back to the edit url /my-pipeline/876108b8-32ac-47b3-b641-c4089447a4bc/edit and check if the values have been updated.

## Screenshots
![image](https://user-images.githubusercontent.com/5575331/82442713-179b9480-9a98-11ea-9f8b-07177259f931.png)


[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
